.github/workflows/clients-dotnet.yml

```diff
@@ -1,242 +0,0 @@
-        with:
-          name: test-project-artifact
-          path: ./test-project
-      - name: Test nuget package
-        working-directory: ./test-project
-        run: dotnet run
-
-  run_validation_tests_on_containers:
-    if: inputs.version == ''
-    needs: prepare_validation_tests
-    name: Validate dotnet runtime
-    strategy:
-      fail-fast: false
-      matrix:
-        # Min supported, LTS and latest versions, gnu and musl.
-        tag:
-          - "7.0"
-          - "6.0"
-          - "3.1"
-          - "7.0-alpine"
-          - "6.0-alpine"
-          - "3.1-alpine"
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/download-artifact@v3
-        with:
-          name: nuget-artifact-${{ inputs.version }}
-          path: ./nuget
-      - uses: actions/download-artifact@v3
-        with:
-          name: test-project-artifact
-          path: ./
-      - name: Test nuget package
-        run: |
-          docker run -v $(pwd)/nuget:/nuget -v $(pwd)/Program.cs:/Program.cs mcr.microsoft.com/dotnet/sdk:${{ matrix.tag }} sh -c "
-          mkdir test-project && cd test-project
-          dotnet nuget add source /nuget
-          dotnet new console
-          dotnet add package tigerbeetle -s /nuget
-          cp -f /Program.cs .
-          dotnet run
-          "
-
-  publish:
-    if: inputs.version != ''
-    needs: build
-    name: Publish NuGet package
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/download-artifact@v3
-        with:
-          name: nuget-artifact-${{ inputs.version }}
-          path: ./nuget
-      - run: |
-          dotnet nuget push ./nuget/*.nupkg \
-          --api-key $NUGET_KEY \
-          --source https://api.nuget.org/v3/index.json
-        env:
-          NUGET_KEY: ${{ secrets.NUGET_KEY }}
-
-  samples_integration_test:
-    if: inputs.version == ''
-
-    # We use self hosted runners for M1 here. See macos.yml for an explaination
-    permissions:
-      contents: read
-
-    needs: build
-
-    strategy:
-      matrix:
-        os: [ubuntu-latest, macos-latest, windows-latest, [self-hosted, ARM64, macos-12.6], [self-hosted, ARM64, macos-13.2]]
-        sample: [basic, two-phase, two-phase-many]
-
-    runs-on: ${{ matrix.os }}
-    steps:
-      - uses: actions/checkout@v3
-
-      - uses: actions/setup-dotnet@v3
-        with:
-          # Min supported and latest versions.
-          dotnet-version: |
-            3.1.x
-            7.0.x
-
-      - uses: actions/download-artifact@v3
-        with:
-          name: nuget-artifact-${{ inputs.version }}
-          path: ./nuget
-
-      # Grab Zig
-      - run: ./scripts/install.${{ matrix.os == 'windows-latest' && 'bat' || 'sh' }}
-
-      # Build and run the integration tests
-      - run: ./zig/zig build client_integration
-      - run: ./zig-out/bin/client_integration${{ matrix.os == 'windows-latest' && '.exe' || '' }} --language=dotnet --sample=${{ matrix.sample }}
-
-  docs:
```

Comment:
I keep the client docs setup mostly as is. I do want to fold that into `client_ci`, but that's going to be a follow  up.